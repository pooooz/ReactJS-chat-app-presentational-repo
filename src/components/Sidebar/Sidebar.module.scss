.sidebar {
  font-family: 'Press Start 2P', cursive;
  min-height: 929px;
  min-width: 250px;
  margin: 0;
  padding: 0;
  ul {
    list-style: none;
    padding: 0 0 0 20px;
  }
  background-color: rgba(46, 66, 7, 0.1);
  color: limegreen;

  &__element {
    margin: 10px 0 0 0;

  }

  &__link {
    color: limegreen;
  }

  &__button {
    margin: 30px 0 0 0;
    padding: 5px;
    font-family: 'Press Start 2P', cursive;
    color: limegreen;
    cursor: pointer;
    background: transparent;
    border: 1px solid limegreen;
  }
}

.main {
  width: 100%;
}

.flex_wrap {
  width: 100%;
  min-height: 929px;
  display: flex;
}