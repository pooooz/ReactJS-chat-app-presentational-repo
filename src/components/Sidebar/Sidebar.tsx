import React, { FC } from 'react';
import { NavLink, Outlet } from 'react-router-dom';

import styles from './Sidebar.module.scss';

export const Sidebar: FC = () => {
  return (
    <>
      <aside className={styles.sidebar}>
        <ul>
          <li className={styles.sidebar_element}>
            <NavLink className={styles.link} to="/profile">
              Profile
            </NavLink>
          </li>
          <li className={styles.sidebar_element}>
            <NavLink className={styles.link} to="/chats">
              Messages
            </NavLink>
          </li>
        </ul>
      </aside>
      <main className={styles.main}>
        <Outlet />
      </main>
    </>
  );
};
