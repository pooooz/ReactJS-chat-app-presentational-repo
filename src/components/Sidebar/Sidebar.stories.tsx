import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Sidebar } from './Sidebar';
import { MemoryRouter } from 'react-router-dom';

export default {
  title: 'components/Sidebar',
  component: Sidebar,
} as ComponentMeta<typeof Sidebar>;

export const Default: ComponentStory<typeof Sidebar> = () => (
  <MemoryRouter>
    <Sidebar />
  </MemoryRouter>
);
