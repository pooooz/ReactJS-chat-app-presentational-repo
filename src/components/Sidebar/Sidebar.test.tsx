import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/user-event';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { Sidebar } from './Sidebar';

describe('Sidebar', () => {
  const history = createMemoryHistory();
  it('Render component', () => {
    const { asFragment } = render(
      <Router location="/" navigator={history}>
        <Sidebar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
