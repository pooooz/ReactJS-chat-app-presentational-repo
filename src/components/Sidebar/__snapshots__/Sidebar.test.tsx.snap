// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Render component 1`] = `
<DocumentFragment>
  <aside
    class="sidebar"
  >
    <ul>
      <li
        class="sidebar_element"
      >
        <a
          class="link"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="sidebar_element"
      >
        <a
          class="link"
          href="/chats"
        >
          Messages
        </a>
      </li>
    </ul>
  </aside>
  <main
    class="main"
  />
</DocumentFragment>
`;
